src/mono/browser/runtime/managed-exports.ts

```diff
@@ -68,7 +71,7 @@ export async function call_entry_point(entry_point: MonoMethod, program_args?: s
             promise = Promise.resolve(0);
         }
         (promise as any)[do_not_force_dispose] = true; // prevent disposing the task in forceDisposeProxies()
-        return await promise;
+        return promise;
```

Comment:
This changes when the finally is run. What if the promise asynchronously fail?

Reply:
it was wrong from perspective of `sp` - stack pointer. I improved it again.